sunpy/net/dataretriever/tests/test_norh.py

```diff
@@ -16,26 +20,107 @@
+    """
+    # Create a mock QueryResponse object
+    map_ = {
+        'TimeRange': TimeRange(parse_time(start_date), parse_time(end_date)),
+        'Time_start': parse_time(start_date),
+        'Time_end':  parse_time(end_date),
+        'source': 'NAOJ',
+        'instrument': 'NORH',
+        'physobs': '',
+        'provider': 'NRO',
+        'wavelength': wavelength,
+    }
+
+    # Create a range of Time
+    st_datetime = datetime.strptime(start_date, '%Y/%m/%d')
+    ed_datetime = datetime.strptime(end_date, '%Y/%m/%d') + timedelta(days=1)
+    time_range = TimeRange(st_datetime, ed_datetime).split(int((ed_datetime-st_datetime).days))
+
+    with mock.patch('sunpy.net.dataretriever.sources.norh.NoRHClient._get_url_for_timerange',
+                    return_value=create_url(start_date, end_date, wavelength)):
+        resp = QueryResponse.create(map_,
+        NORHClient._get_url_for_timerange(map_['TimeRange'], wavelength=wavelength), time=time_range)
+        # Attach the client with the QueryResponse
+        assert resp
+        resp.client = NORHClient
+        return resp
+
+
 @pytest.mark.remote_data
-@pytest.mark.parametrize("timerange,url_start,url_end", [
-    (TimeRange('2012/4/21', '2012/4/21'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/04/tca120421',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/04/tca120421'
-     ),
-    (TimeRange('2012/12/1', '2012/12/2'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/12/tca121201',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/12/tca121202'
-     ),
-    (TimeRange('2012/3/7', '2012/3/14'),
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120307',
-     'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120314'
-     )
-])
-def test_get_url_for_time_range(timerange, url_start, url_end):
-    urls = norh.NoRHClient()._get_url_for_timerange(timerange, wavelength=17*u.GHz)
+def test_fetch_working():
+    """
+    Tests if the mock fetch is containing the correct data.
+    """
+
+    qr1 = NORHClient.search(a.Time('2012/10/4', '2012/10/6'), a.Instrument('norh'), a.Wavelength(17*u.GHz))
+
+    # Create a mock query object
+    mock_qr = mock_query_object('2012/10/4', '2012/10/6', wavelength=17*u.GHz)
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+    # Assert the values
+    assert mock_qr.source == qr.source
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.url == qr.url
+    assert mock_qr.time == qr.time
+
+    # Assert if the time range is same
+    assert qr1.time_range() == TimeRange('2012/10/4', '2012/10/7')
+
+    # Assert the fetch object, and whether it returns the correct set of files
+    res = NORHClient.fetch(qr1)
```

Comment:
What happens if you use the mock_qr and pass it into fetch?

Reply:
Do you expect some certaing behaviour, in particular some error, as I used mock_qr, and it returned a Result object as expected, correctly ?

Reply:
I would kind of expect it to work?

Reply:
I have tried that, and it works.